crates/polars-lazy/src/physical_plan/executors/scan/ipc.rs

```diff
@@ -12,14 +17,46 @@ pub struct IpcExec {
 
 impl IpcExec {
     fn read(&mut self, verbose: bool) -> PolarsResult<DataFrame> {
-        let file = std::fs::File::open(&self.path)?;
+        if is_cloud_url(&self.path) || env_force_async() {
+            #[cfg(not(feature = "cloud"))]
+            {
+                panic!("activate cloud feature")
+            }
+
+            #[cfg(feature = "cloud")]
+            {
+                // TODO: This will block the current thread until the data has been
```

Comment:
This is fine as the default engine cannot work on batches anyway. What our goal here will be is downloading parts of the file (e.g. `Recordbatches/ Chunks`) concurrently.